test_ros2cli/package.xml

```diff
@@ -12,11 +12,21 @@
   <test_depend>action_tutorials</test_depend>
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
+  <test_depend>demo_nodes_py</test_depend>
```

Comment:
This will create a repository level cycle since packages in the repository containing `demo_nodes_py` depend on packages in this repo. Imo we shouldn't do that.

Reply:
Ok, that is true for `ros2run`. It's another instance of an issue we have elsewhere e.g. `launch_ros`, and that I was consciously eluding to avoid scope creep. IMO we shouldn't be using demo nodes for system or integration testing, neither duplicate `ros2/system_tests` talkers, listeners, servers, etc., and others. We need a package with test nodes that `ros2/ros2cli`, `ros2/launch*`, `ros2/system_tests`, can all depend on. Probably in its own repo, like we did for `test_interface_files`.

I can put this on hold, do that and come back, or fix it afterwards. What would you rather do?

Reply:
Imo a package/repo like `test_nodes` sounds like a good idea.